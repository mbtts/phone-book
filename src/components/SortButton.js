import React, { PureComponent } from "react";

import PropTypes from "prop-types";
import SVGInline from "react-svg-inline";
import arrowDropDown from "../../assets/svg/round-arrow_drop_down-24px.svg";
import arrowDropUp from "../../assets/svg/round-arrow_drop_up-24px.svg";

class SortButton extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const asc = this.props.order === "asc";
    const label = asc ? "Asc" : "Desc";
    const icon = asc ? arrowDropUp : arrowDropDown;
    return (
      <button onClick={this.props.onClick} className={this.props.className}>
        <SVGInline width="24" svg={this.props.icon} />
        <span>{label}</span>
        <SVGInline width="24" svg={icon} />
      </button>
    );
  }
}

SortButton.propTypes = {
  icon: PropTypes.string.isRequired,
  order: PropTypes.oneOf(["asc", "desc"]).isRequired,
  onClick: PropTypes.func.isRequired,
  className: PropTypes.string.isRequired
};

export default SortButton;
