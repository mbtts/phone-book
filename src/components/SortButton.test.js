import React from "react";
import SortButton from "./SortButton";

describe("Sort button", () => {
  it("should render ascending", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <SortButton
        className="sortButton"
        icon="<svg />"
        order="asc"
        onClick={onClick}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button span").text()).toEqual("Asc");
  });

  it("should render descending", () => {
    const onClick = jest.fn();
    const wrapper = shallow(
      <SortButton
        className="sortButton"
        icon="<svg />"
        order="desc"
        onClick={onClick}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("button span").text()).toEqual("Desc");
  });

  it("should invoke the click function when pressed", () => {
    const onClick = jest.fn();

    const wrapper = shallow(
      <SortButton
        className="sortButton"
        icon="<svg />"
        order="desc"
        onClick={onClick}
      />
    );

    wrapper.find("button").simulate("click");
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
