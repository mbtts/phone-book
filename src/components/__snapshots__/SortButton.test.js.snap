// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort button should render ascending 1`] = `
<button
  className="sortButton"
  onClick={[MockFunction]}
>
  <SVGInline
    classSuffix="-svg"
    cleanup={Array []}
    cleanupExceptions={Array []}
    component="span"
    svg="<svg />"
    width="24"
  />
  <span>
    Asc
  </span>
  <SVGInline
    classSuffix="-svg"
    cleanup={Array []}
    cleanupExceptions={Array []}
    component="span"
    svg="<svg />"
    width="24"
  />
</button>
`;

exports[`Sort button should render descending 1`] = `
<button
  className="sortButton"
  onClick={[MockFunction]}
>
  <SVGInline
    classSuffix="-svg"
    cleanup={Array []}
    cleanupExceptions={Array []}
    component="span"
    svg="<svg />"
    width="24"
  />
  <span>
    Desc
  </span>
  <SVGInline
    classSuffix="-svg"
    cleanup={Array []}
    cleanupExceptions={Array []}
    component="span"
    svg="<svg />"
    width="24"
  />
</button>
`;
